pythonforandroid/graph.py

```diff
@@ -99,17 +129,125 @@ def find_order(graph):
                 bset.discard(result)
 
 
-def get_recipe_order_and_bootstrap(ctx, names, bs=None):
-    recipes_to_load = set(names)
+def obvious_conflict_checker(ctx, name_tuples, blacklist=[]):
+    """ This is a pre-flight check function that will completely ignore
+        recipe order or choosing an actual value in any of the multiple
+        choice tuples/dependencies, and just do a very basic obvious
+        conflict check.
+    """
+    deps_were_added_by = dict()
+    deps = set()
+
+    # Add dependencies for all recipes:
+    to_be_added = [(name_tuple, None) for name_tuple in name_tuples]
+    while len(to_be_added) > 0:
+        current_to_be_added = list(to_be_added)
+        to_be_added = []
+        for (added_tuple, adding_recipe) in current_to_be_added:
+            if len(added_tuple) > 1:
+                # No obvious commitment in what to add, ignore it.
+                # (Remember this function only catches obvious issues)
+                continue
+
+            name = added_tuple[0]
+            recipe_conflicts = []
+            recipe_dependencies = []
+            try:
+                # Get recipe to add and who's ultimately adding it:
+                recipe = Recipe.get_recipe(name, ctx)
+                recipe_conflicts = [c.lower() for c in recipe.conflicts]
+                recipe_dependencies = \
+                    get_dependency_tuple_list_for_recipe(
+                        recipe, blacklist=blacklist
+                    )
+            except ValueError:
+                pass
+            adder_recipe_1_name = adding_recipe or name
+
+            # See if this recipe itself conflicts with what was added before:
+            triggered_conflicts = []
+            for dep_tuple_list in deps:
+                if len(set(dep_tuple_list).intersection(
+                        set(recipe_conflicts))) == len(dep_tuple_list):
+                    triggered_conflicts.append(dep_tuple_list)
+
+            # See if what was added before conflicts with this recipe:
+            for dep_tuple_list in deps:
+                if len(dep_tuple_list) > 1:
+                    # Not an obvious commitment, so we won't check.
+                    # (remember this function only catches obvious issues)
+                    continue
+                try:
+                    conflicts = [c.lower() for c in
+                                 Recipe.get_recipe(dep_tuple_list[0], ctx).
+                                 conflicts]
+                    if name in conflicts:
+                        triggered_conflicts.append(dep_tuple_list)
+                except ValueError:
+                    pass
+
+            # Throw error on conflict:
+            if len(triggered_conflicts) > 0:
+                # Get first conflict and see who added that one:
+                adder_recipe_2_name = "'||'".join(triggered_conflicts[0])
+                adder_recipe_2_name =\
```

Comment:
Python is not bash :)

Reply:
@KeyWeeUsr again, looks IMHO considerably more cramped and less readable without it (I'm hot adding these for fun), but I'm happy to change it. also, would you mind properly writing out the remark next time? I don't think that is the most obvious or pleasant way to word it, but maybe that's just me :grimacing:

Reply:
@KeyWeeUsr I removed this backslash now by just breaking apart the code a little more. I was probably just trying to do too much in a single line :man_shrugging:

Reply:
resolving since the backslash use is gone :sunglasses: :+1: